doc/TAM/SAI-Advanced-Telemetry.md

```diff
@@ -0,0 +1,1002 @@
+  
+attr_count = 2;
+ 
+sai_create_tam_transport_fn(
+    &sai_tam_transport_obj,
+    switch_id,
+    attr_count,
+    sai_attr_list);
+    
+/*
+ * Transport type can be specified as port, to directly send 
+ * reports out on a management interface
+ */
+
+sai_attr_list[0].id = SAI_TAM_TRANSPORT_ATTR_TRANSPORT_TYPE; 
+sai_attr_list[0].value.s32 = SAI_TAM_TRANSPORT_TYPE_PORT;
+
+sai_attr_list[1].id = SAI_TAM_TRANSPORT_ATTR_MTU;
+sai_attr_list[1].value.u32 = 1500;
+
+sai_attr_list[2].id = SAI_TAM_TRANSPORT_ATTR_SRC_MAC_ADDRESS;
+sai_attr_list[2].value.u32 = 1500;
+
+sai_attr_list[3].id = SAI_TAM_TRANSPORT_ATTR_DST_MAC_ADDRESS;
+sai_attr_list[3].value.u32 = 1500;
+
+attr_count = 4;
+ 
+sai_create_tam_transport_fn(
+    &sai_tam_transport_obj,
+    switch_id,
+    attr_count,
+    sai_attr_list);
+ 
+```
+##### Step 4: Create a Collector Object
+
+```sh
+sai_attr_list[0].id = SAI_TAM_COLLECTOR_ATTR_SRC_IP;
+sai_attr_list[0].value.ipaddr.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+sai_attr_list[0].value.ipaddr.ip4 = 0x0101010a; 
+ 
+sai_attr_list[1].id = SAI_TAM_COLLECTOR_ATTR_DST_IP;
+sai_attr_list[1].value.ipaddr.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+sai_attr_list[1].value.ipaddr.ip4 = 0x0101010b;
+ 
+sai_attr_list[2].id = SAI_TAM_COLLECTOR_ATTR_TRANSPORT; 
+sai_attr_list[2].value.oid = sai_tam_transport_obj;
+ 
+attr_count = 3;
+    sai_create_tam_collector_fn(
+    &sai_tam_collector_obj,
+    switch_id,
+    attr_count,
+    sai_attr_list);
+```
+##### Step 5: Create an Event  Object
+```sh
+sai_attr_list[0].id = SAI_TAM_EVENT_ATTR_TYPE;
+sai_attr_list[0].value.s32 = SAI_TAM_EVENT_TYPE_PACKET_DROP;
+
+sai_attr_list[1].id = SAI_TAM_EVENT_ATTR_PACKET_DROP_TYPE;
+sai_attr_list[1].value.s32 = SAI_PACKET_DROP_TYPE_INGRESS;
+
+sai_attr_list[2].id = SAI_TAM_EVENT_ATTR_ACTION_LIST;
+sai_attr_list[2].value.objlist.count = 1;
+sai_attr_list[2].value.objlist.list[0] = sai_tam_event_action_obj;
+
+sai_attr_list[3].id = SAI_TAM_EVENT_ATTR_COLLECTOR_LIST;
+sai_attr_list[3].value.objlist.count = 1;
+sai_attr_list[3].value.objlist.list[0] = sai_tam_collector_obj;
+
+attr_count = 4;
+
+sai_create_tam_event_fn(
+    &sai_tam_event_packet_drop_obj,
+    switch_id,
+    attr_count,
+    sai_attr_list);
+```
+##### Step 7: Create TAM Object
+```sh
+sai_attr_list[0].id = SAI_TAM_ATTR_EVENT_OBJECTS_LIST;
+sai_attr_list[0].value.objlist.count = 1;
+sai_attr_list[0].value.objlist.list[0] = sai_tam_event_packet_drop_obj;
+sai_attr_list[1].id = SAI_TAM_ATTR_TAM_BIND_POINT_TYPE_LIST;
+sai_attr_list[1].value.objlist.count = 2;
+sai_attr_list[1].value.objlist.list[0] = SAI_TAM_BIND_POINT_TYPE_PORT;
+sai_attr_list[1].value.objlist.list[1] = SAI_TAM_BIND_POINT_TYPE_QUEUE;
+
+attr_count = 2;
+
+sai_create_tam_fn(
+    &sai_tam_obj,
+    switch_id,
+    attr_count,
+    sai_attr_list);
+```
+##### Step 7.1 : Event forwarding to Port
+Binding TAM object to port object for all events will result in forwarding for reports to the port if transport type is specified as TYPE_PORT

```

Comment:
It's kind of misleading to bind a tam object to port just for sending report out of the port.
Could we just specify the report port in tam transport object?